modules/services/window-managers/i3.nix

```diff
@@ -772,20 +772,10 @@ in
       home.packages = [ cfg.package ];
       xsession.windowManager.command = "${cfg.package}/bin/i3";
       xdg.configFile."i3/config".source = configFile;
-
-      home.activation.checkI3 = dag.entryBefore [ "linkGeneration" ] ''
-        if ! cmp --quiet \
-            "${configFile}" \
-            "${config.xdg.configHome}/i3/config"; then
-          i3Changed=1
-        fi
-      '';
-
-      home.activation.reloadI3 = dag.entryAfter [ "linkGeneration" ] ''
-        SOCKET=''${XDG_RUNTIME_DIR:-/run/user/$UID}/i3/ipc-socket.*
-        if [ -v i3Changed ] && [ -S $SOCKET ]; then
+      xdg.configFile."i3/config".onChange = ''
+        if [[ -v DISPLAY ]]; then
           echo "Reloading i3"
-          ${cfg.package}/bin/i3-msg -s $SOCKET reload 1>/dev/null
+          $DRY_RUN_CMD ${cfg.package}/bin/i3-msg reload 1>/dev/null
```

Comment:
It's safe to leave out the socket here?

@uvNikita Do you know if this could cause issues?

Reply:
Specifying socket explicitly is necessary to fix reloading when home-manager is used as nixos module since in this case `i3-msg` command can't find it automatically. I tested this branch and i3 is not reloading for me on config changes, adding socket related code back fixes the issue.

@ayyjayess was there a specific reason for removing this lines?

Reply:
@uvNikita I've reinserted the socket (with some slight modifications) in my onChange branch: https://github.com/rycee/home-manager/blob/9874354cb297a853c64b67e1098b09a2c666fc7b/modules/services/window-managers/i3.nix#L776-L782
Seems OK?

Reply:
@uvNikita I must have done a bad merge wrong when bringing this PR up to date.

Reply:
@rycee it looks good, except that it should be `if [ -S $i3Socket ]; then`, it doesn't work with double brackets for some reason.

Reply:
@uvNikita Ah, I didn't know that `-S` didn't work with `[[`. I've updated the branch and merged to master now 🙂